### PR TITLE
[GOVCMSD10-232] New event subscriber

### PR DESCRIPTION
**Pull request by mingsong-hu**
This PR introduces a 'HTTP Request' event handler which will be triggered on every main HTTP request.

Inside the handler function, it will check all files' extensions uploaded via that request. If a blocked extension detected, an 'Access Denied' exception will be thrown and the file upload process will be terminated.

### CODE REVIEW

File: modules/custom/core/govcms_security/src/GovcmsFileConstraintInterface.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Provides an interface for constrants on files uploaded.
+ */
+interface GovcmsFileConstraintInterface {
+
+  /**
+   * A list of blocked file extensions.
+   */
+  public const BLOCKED_EXTENSIONS = ['doc', 'xls', 'ppt', 'rtf'];
```
Comment (larowlan) [inline modules/custom/core/govcms_security/src/GovcmsFileConstraintInterface.php:13]: should this also be extended to other malicious file extensions like .exe etc?

**Review by larowlan — COMMENTED** [review]
I think this should come with a browser test or some form of test coverage

You should be able to test this in isolation by extending from BrowserTestBase and using one of the test modules in core to expose a file upload field - eg file_test

File: modules/custom/core/govcms_security/src/GovcmsFileConstraintInterface.php (hunk 1 above)
Comment (mingsong-hu) [reply modules/custom/core/govcms_security/src/GovcmsFileConstraintInterface.php:13]: Thanks @larowlan. :heart:

**Review by larowlan — COMMENTED** [review]
nice - I was expecting you could use the test modules from core rather than need to write your own, but whatever works for you :+1:

**Comment by mingsong-hu** [comment]
> nice - I was expecting you could use the test modules from core rather than need to write your own, but whatever works for you 👍

Our requirement (expected result) is different from core. Any insecure file extension will be rename(extension) to txt as designed result. In our case, we don't rename it but halt the uploading process instead.  So the expected result of test is not the same.  Also, we don't want to run those irrelevant tests from file_test module from the core.
@larowlan , Thank you so much for your help.

**Review by pandaskii — APPROVED** [review]
Looks good

**Pull request merged**